### DISCUSSION TITLE
Feature request: option to show a standard menu bar on Linux

### DISCUSSION THREAD

**Discussion by unikitty37** [ghostty-org/ghostty#3434] [Feature Requests, Ideas]
Ghostty uses a hamburger menu, but this is only accessible if `gtk-titlebar` is true. If `gtk-titlebar` is false, the menu isn't shown, and can't be opened via a keybinding as far as I can see.

I set `gtk-titlebar` to false because I'm using KDE, and the GTK titlebar looks extremely out of place. But without the menu, I have to memorise every keybinding — I can't go to the menu for commands I can't remember the bindings for (or see the bindings against the menu options and learn that way :)

It would be great if there was an option to replace the hamburger menu with a normal menu bar.

**Comment by mitchellh** [comment]
Do you have screenshots of what you're proposing? (i.e. from another app)

**Reply by tomektom** [reply]
It could be done like in Konsole. You have menu under right click (and many other options, too many in my opinion)
![image](https://github.com/user-attachments/assets/ae03a3ac-38fb-4c0a-be4d-5f268b2d021d)

**Reply by unikitty37** [reply]
> Do you have screenshots of what you're proposing? (i.e. from another app)

This is what I was thinking of — just a standard menubar in the standard place that works in the standard way that everybody's used to :)

![image](https://github.com/user-attachments/assets/6e43e42c-3b45-4762-9213-b502d040bb7a)

(though a way of having the hamburger menu display when not using GTK titlebars would work as well, but I'm not sure where you'd put it if you can't tweak the titlebar…)

**Reply by iamgreaser** [reply]
For comparison:

* KDE konsole: File, Edit, View, Bookmarks, Plugins, Settings, Help (from the screenshot provided by unikitty37)
* xfce4-terminal: &File, &Edit, &View, &Terminal, T&abs, &Help (using "&" as a prefix for which menu item is selected when one does Alt + letter)

I'm mixed as to whether "File" is the right word to use, but I'm not about to bikeshed an alternative name for it. Plus in theory it makes localisation easier to leave it as "File".

The de-facto standard way to arrange things is &File (or equivalent), &Edit, &View, insert any extra menu items here, &Help.

(On Unix systems, &Help is traditionally right-aligned while the rest are left-aligned, but I think that is a lot less common nowadays - none of the programs I have open right now seem to special-case &Help in this manner, and while Windows does have built-in support for this arrangement, it seems to have been abandoned by the time Windows 3.1 came out.)

**Reply by tomektom** [reply]
> The de-facto standard way to arrange things is &File (or equivalent), &Edit, &View, insert any extra menu items here, &Help.

That's probably more dependent of how specific DE implements this functionality. On both KDE and Hyprland I see underline eg. <ins>F</ins>ile, <ins>E</ins>dit - for me that's definitely more common than &

**Reply by 00-kat** [reply]
In Qt you can put a & before it in your code to underline it, and use <kbd>alt</kbd>+<kbd>\<underlined key\></kbd> to select that menu item with a keyboard.  Not sure about other toolkits.

**Reply by iamgreaser** [reply]
> On both KDE and Hyprland I see underline eg. File, Edit - for me that's definitely more common than &

Yes, it's underlined for XFCE too. It's a bit of a programmery way to denote which character is underlined. Sorry if it wasn't clear.

**Reply by pluiedev** [reply]
> Not sure about other toolkits.

In GTK they're called ["mnemonics"](https://docs.gtk.org/gtk4/ctor.Label.new_with_mnemonic.html)

**Comment by jcollie** [comment]
There is already a right-click context menu, although there are fewer entries in the context menu than in the normal menu.

**Reply by mitchellh** [reply]
I think we talked about this before, and not sure if it would satisfy the OP of this issue, but if the titlebar/headerbar is hidden, it would be nice to put the entire menu in the context menu under a "Menu" item.

**Reply by unikitty37** [reply]
I'd prefer a normal menubar as it can also be navigated entirely by keyboard, but adding a "Menu" item to the context menu would certainly be an improvement on the current behaviour where the menu items aren't available at all without having the GTK titlebar.

**Comment by gtxaspec** [comment]
Is there a way to reduce the title bar height? The current default is quite thick (see screenshot).

I've tried editing the following GTK theme files without success:
```
~/.config/gtk-4.0/gtk.css
~/.config/gtk-3.0/gtk.css
~/.config/libadwaita-1/gtk.css
```

![image](https://github.com/user-attachments/assets/43e992aa-ce60-407c-9209-45a239268b3f)

**Reply by 00-kat** [reply]
Both [`gtk-adwaita`] and https://github.com/ghostty-org/ghostty/discussions/3158#discussioncomment-11685100 work, and this isn't relevant to this discussion.

[`gtk-adwaita`]: https://ghostty.org/docs/config/reference#gtk-adwaita

**Comment by 00-kat** [comment]
I remember this being discussed before, relevant links in (hopefully) chronological order: https://discord.com/channels/1005603569187160125/1005603569711452192/1303113109010907252 https://discord.com/channels/1005603569187160125/1005603569711452192/1305222734145523813 https://discord.com/channels/1005603569187160125/1005603569711452192/1303082183287640125.


I found https://discord.com/channels/1005603569187160125/1005603569711452192/1305242241962348586
>  Opened a discussion on GH

And also #2642.  However #2642 is on the tenth of November (my time) but that Discord message was on the eleventh (my time), so I'm not sure which discussion it was.

**Reply by unikitty37** [reply]
Ah, well, I'm not on Discord as they require my mobile phone number, which they are not getting. So I wouldn't have seen that.

I think #2642 doesn't really cover the issue, as it doesn't mention most of the menu items becoming inaccessible if `gtk-titlebar` is false (which it would be if you're not using Gnome, as it makes the window look completely different from everything else 😁)

**Reply by jcollie** [reply]
#4952 will make the window menu and the context menus identical.

**Reply by andrejdaskalov** [reply]
So, to clear up #2642: We had a discord discussion on making a global menu available, but I ended up concluding it's no longer something you can do with GTK4. I ended up with a traditional menubar demo instead. If there is renewed interest for a traditional menubar inside the window, I could certainly do that. I'm guessing in this case, we could provide it as an optional config option?

**Reply by 00-kat** [reply]
Doesn't Plasma automagically grab a traditional menu bar and make it a global menu bar if you add that widget?  Or had you tried to make your own custom menu bar which looks like the one you normally find?  (Sorry about the wording of this comment, I'm not sure how best to describe what I mean.)

**Reply by andrejdaskalov** [reply]
I'll have to check what I did to be precise, but it was almost certainly a standard menu in GTK. The documentation does seem to mention this: 
> Depending on the desktop environment, this may appear at the top of each window, or at the top of the screen.


So in theory it should do what you said, but I had no luck getting it to behave back when I was making this. I'll definitely give it another try later, to  make sure it wasn't just a bug on my other system.
